controllers/storagecluster/storageconsumer.go

```diff
@@ -0,0 +1,40 @@
+package storagecluster
+
+import (
+	ocsv1 "github.com/red-hat-storage/ocs-operator/api/v4/v1"
+	ocsv1a1 "github.com/red-hat-storage/ocs-operator/api/v4/v1alpha1"
+	"github.com/red-hat-storage/ocs-operator/v4/controllers/defaults"
+	"github.com/red-hat-storage/ocs-operator/v4/controllers/util"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+)
+
+type storageConsumer struct{}
+
+var _ resourceManager = &storageConsumer{}
+
+const localStorageConsumerName = "storageconsumer-local"
+
+func (s *storageConsumer) ensureCreated(r *StorageClusterReconciler, storageCluster *ocsv1.StorageCluster) (ctrl.Result, error) {
+	storageConsumer := &ocsv1a1.StorageConsumer{}
+	storageConsumer.Name = localStorageConsumerName
```

Comment:
We need to make sure that we are not creating this in 4.18 provider mode clusters

Reply:
done.